src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -238,6 +230,140 @@ g.test('sampleCount')
     }, !_success);
   });
 
+g.test('texture_size,1d_texture')
+  .desc(`Test texture size requirement for 1D texture`)
+  .subcases(() =>
+    params()
+      .combine(poptions('format', kAllTextureFormats))
+      .combine(
+        poptions('width', [
+          DefaultLimits.maxTextureDimension1D - 1,
+          DefaultLimits.maxTextureDimension1D,
+          DefaultLimits.maxTextureDimension1D + 1,
+        ])
+      )
+      .combine(poptions('height', [1, 2]))
+      .combine(poptions('depthOrArrayLayers', [1, 2]))
+  )
+  .fn(async t => {
+    const { format, width, height, depthOrArrayLayers } = t.params;
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const descriptor: GPUTextureDescriptor = {
+      size: [width, height, depthOrArrayLayers],
+      dimension: '1d' as const,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+
+    const success =
+      width <= DefaultLimits.maxTextureDimension1D && height === 1 && depthOrArrayLayers === 1;
+
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
+  });
+
+g.test('texture_size,2d_texture')
+  .desc(`Test texture size requirement for 2D texture`)
+  .subcases(() =>
+    params()
+      .combine(poptions('format', kAllTextureFormats))
+      .combine(
+        poptions('width', [
+          DefaultLimits.maxTextureDimension2D - 1,
+          DefaultLimits.maxTextureDimension2D,
+          DefaultLimits.maxTextureDimension2D + 1,
+        ])
+      )
+      .combine(
+        poptions('height', [
+          DefaultLimits.maxTextureDimension2D - 1,
+          DefaultLimits.maxTextureDimension2D,
+          DefaultLimits.maxTextureDimension2D + 1,
+        ])
+      )
+      .combine(
+        poptions('depthOrArrayLayers', [
+          DefaultLimits.maxTextureArrayLayers - 1,
+          DefaultLimits.maxTextureArrayLayers,
+          DefaultLimits.maxTextureArrayLayers + 1,
+        ])
+      )
```

Comment:
It turned out that width*height*depthOrArrayLayers will lead to out-of-memory issue. I remember that we don't have any limit about the entire texture memory footprint (width*height*dpehtOrArrayLayers). Right? So I think I need to change other parameters to 1 when I test the bound of each parameter.

Reply:
Since these are just validation tests, we don't need to test create max * max * max sized textures. We're only concerned here with making sure the implementation enforces the maximum exactly. So you could just have 9 cases instead of 3 * 3 * 3:

{ width: max - 1, height: 1, depth: 1 }
{ width: max, height: 1, depth: 1 }
{ width: max + 1, height: 1, depth: 1 }
{ width: 1, height: max - 1, depth: 1 }
...

Reply:
Exactly. I am doing so right now.

Reply:
Another problem is the alignment issue for compressed texture: max - 1 is valid for uncompressed texture but it is invalid for compressed texture . I just skipped compressed formats and added a TODO. I will submit separate PR soon.

Reply:
Sounds good. Maybe the way to do this is parameterize over "diff from max in blocks" and test -1,0,1 and compute the actual size inside the test.

Reply:
Yep.